pkg/controller/operators/catalog/subscription/reconciler.go

```diff
@@ -109,6 +121,104 @@ func (c *catalogHealthReconciler) Reconcile(ctx context.Context, in kubestate.St
 	return
 }
 
+// updateDeprecatedStatus adds deprecation status conditions to the subscription when present in the cache entry
+func updateDeprecatedStatus(sub *v1alpha1.Subscription, time *metav1.Time, client clientv1alpha1.SubscriptionInterface, source map[cache.SourceKey]cache.Source) error {
+	changed := false
+	rollupMessage := ""
+	sourceKey := cache.SourceKey{
+		Name:      sub.Spec.CatalogSource,
+		Namespace: sub.Namespace,
+	}
+	entries, err := source[sourceKey].Snapshot(context.Background())
```

Comment:
Nit: to avoid the potentially confusing `entries.Entries` usage below and from the `.Snapshot()` function returning `(cache.Snapshot, err)` instead of `([]cache.Entry, err)`
```suggestion
	snapshot, err := source[sourceKey].Snapshot(context.Background())
```